frontend/packages/metal3-plugin/src/components/host-menu-actions.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { asAccessReview, KebabOption } from '@console/internal/components/utils';
+import { k8sKill, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { getMachineNodeName, getName } from '@console/shared';
+import { confirmModal } from '@console/internal/components/modals';
+import { NodeMaintenanceModel } from '@console/kubevirt-plugin/src/models';
+import { findNodeMaintenance, getHostMachine, getNodeMaintenanceReason } from '../selectors';
+import { startNodeMaintenanceModal } from './modals/start-node-maintenance-modal';
+
+type ActionArgs = {
+  nodeName?: string;
+  nodeMaintenance?: K8sResourceKind;
+  hasNodeMaintenanceCapability: boolean;
+};
+
+export const SetNodeMaintanance = (
+  kindObj: K8sKind,
+  host: K8sResourceKind,
+  resources: null,
```

Comment:
Just wondering, isn't this originally a (possibly empty) array? And why don't we leverage `resources` here, is it because we need just `host` with additional capabilities passed via custom data?

Reply:
> Just wondering, isn't this originally a (possibly empty) array?

it is an object which is not needed: fixed  by replaced with `{}` ?

>  And why don't we leverage resources here, is it because we need just host with additional capabilities passed via custom data?

yes and also precomputing the data (e.g. finding nodeMaintenance)